examples/interactive_scripting.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import napari
+import time
+
+
+with napari.gui_qt():
+    # create the viewer with an image
+    data = np.random.random((512, 512))
+    viewer = napari.Viewer()
+    layer = viewer.add_image(data)
+
+    def layer_update(*, update_period, num_updates):
+
+        # number of times to update
+        for k in range(num_updates):
+            time.sleep(update_period)
+
+            dat = np.random.random((512, 512))
+            layer.data = dat
+
+            # check that data layer is properly assigned and not blocked?
+            while layer.data.all() != dat.all():
+                layer.data = dat
+
+    update_thread = viewer.update(layer_update, update_period=0.1, num_updates=100)
```

Comment:
As I understand it, this will now fail our linting from #638, because `update_thread` is never used. You can rename it `_update_thread` and I think it will pass, or maybe it needs to be just `_`. @tlambert03?